src/electron/blocker.ts

```diff
@@ -0,0 +1,36 @@
+/*!
+ * Copyright (c) 2017-2019 Cliqz GmbH. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+// Type definition
+import * as electron from 'electron';
+
+import Engine from '../engine/engine';
+import Request from '../request';
+
+/**
+ * Wrap `FiltersEngine` into a Electron-friendly helper class.
+ */
+export default class ElectronBlocker extends Engine {
+  public enableBlockingInSession(ses: Electron.Session) {
+    ses.webRequest.onBeforeRequest({ urls: ['<all_urls>'] }, this.onRequest);
```

Comment:
I think `this` will be `undefined` from the code in `onRequest`. So maybe we could wrap it in a lambda or bind it? `(...args) => this.onRequest(...args)`

Reply:
It's already done by using transform class property
```typescript
private onRequest = (
    details: Electron.OnBeforeRequestDetails,
    callback: (a: Electron.Response) => void,
  ): void => {
```

Reply:
Ho right! Did not pay attention to this :+1: